File: TP1 python/python fede/ejercicio 2/Ticket.py
import datetime

class Ticket:
    
    def __init__(self, nroVenta):
        self.nroVenta = nroVenta
        self.fecha = datetime.now()
        self.productos = []
        
    def agregarProducto(self, producto):
        self.productos.append(producto)
    
    def calcularTotal(self):
        total = 0.0
        for produ in self.productos:
            total += produ.getPrecio()
        return total
    
    def imprimir(self):
        print("Nro de venta: {}          Fecha: {}".format(self.nroVenta, self.fecha))
        print("Articulo       Dosis          Precio")
        for produ in self.productos:
            produ.imprimir()
        print("                         Total:", self.calcularTotal())

File: TP1/TP1 java/src/java fede/Ejercicio 2/Ticket.java
import java.time.LocalDate;
import java.util.ArrayList;

public class Ticket {
    
    private Integer nroVenta;
    private LocalDate fecha;
    private ArrayList<Producto> productos;
    
    public Ticket(Integer nroVenta){
        this.nroVenta = nroVenta;
        this.fecha = LocalDate.now();
        this.productos = new ArrayList<>();
    }

    public void agregarProducto(Producto producto){
        this.productos.add(producto);
    }

    public void imprimir(){
        System.out.println("Nro de venta: " + this.nroVenta + "         Fecha: " + this.fecha);
        System.out.println("Articulo        Dosis       Precio");
        for(Producto produ:this.productos){
            produ.imprimir();
        }
        System.out.println(                                             "Total: " + this.calcularTotal());
    }

    public Double calcularTotal(){
        Double total = 0.0;
        for(Producto produ:this.productos){
            total += produ.getPrecio();
            //total = total + produ.getPrecio();
        } 
        return total;
    }
}
